lib/gyp/generator/ninja/index.js

```diff
@@ -140,6 +140,7 @@ Ninja.prototype.expand = function expand(p, productDir) {
     p = p.replace(/\$!INTERMEDIATE_DIR/g, intDir);
   }
 
+  p = p.replace(/(\.\.[\\\/])\$\|CONFIGURATION_NAME/g, `$1${this.configDir}`);
```

Comment:
Voodoo
MSVS at least, resolves paths for actions from the dir of the project file.
I think in our case path resolution for actions start at base of build.